src/main/java/picard/illumina/parser/TileMetricsUtil.java

```diff
@@ -75,12 +75,31 @@
     /**
      * Returns the path to the TileMetrics file given the basecalling directory.
      */
-    public static File renderTileMetricsFileFromBasecallingDirectory(final File illuminaRunDirectory, boolean isNovaSeq) {
+    public static File renderTileMetricsFileFromBasecallingDirectory(final File illuminaRunDirectory, Integer numCycles, boolean isNovaSeq) {
+        return fileTileMetricsFile(illuminaRunDirectory, numCycles, isNovaSeq);
+    }
+
+    private static File fileTileMetricsFile(File illuminaRunDirectory, Integer numCycles, boolean isNovaSeq) {
+        Path interOpDir = illuminaRunDirectory.toPath().resolve(INTEROP_SUBDIRECTORY_NAME);
+        final List<Path> pathsToTest = new ArrayList<>();
+
+        pathsToTest.add(interOpDir.resolve(TILE_METRICS_OUT_FILE_NAME));
+
         if (isNovaSeq) {
-            return new File(new File(illuminaRunDirectory, INTEROP_SUBDIRECTORY_NAME + File.separator + CYCLE_CONTAINING_TILE_METRICS), TILE_METRICS_OUT_FILE_NAME);
-        } else {
-            return new File(new File(illuminaRunDirectory, INTEROP_SUBDIRECTORY_NAME), TILE_METRICS_OUT_FILE_NAME);
+            // check cycles in reverse order.
+            pathsToTest.addAll(IntStream.range(1, numCycles + 1).map(i -> 1 + (numCycles - i)).mapToObj(
```

Comment:
I think a `for` loop would be clearer (much as it pains me)

Reply:
I agree, although `addAll()` of `.collect(Collectors.toList())` would be better written as `stream().map(pathsToTest::add)`